avm/res/app-configuration/configuration-store/tests/e2e/defaults/main.test.bicep

```diff
@@ -43,6 +56,17 @@ module testDeployment '../../../main.bicep' = [
     params: {
       name: '${namePrefix}${serviceShort}001'
       location: resourceLocation
+      enablePurgeProtection: true
```

Comment:
This can be a challenge in our testing, as we usually create an delete resources

Reply:
It is working in the test; however, the real failure is the order of operations as it feels the config store gets deleted first then the replica...which throws an error that the parent has already been deleted. I guess would ask how is this being accommodated for other resources who have purge protection such as Key Vault and Log Analytics?



I am not high on it being the default as it is not available in free tier and not sure on repeat logic as there already is logic to not allow if  `sku=='Free'` .I would suggest keeping it as false so to easily accommodate for more flexibility. 

`enablePurgeProtection: sku == 'Free' ? false : enablePurgeProtection`

Reply:
I totally see the angle @JFolberth. And believe me a wormhole likely opens every time the discussion flexibility/usability vs security is raised 😄 
When it comes to WAF and especially reliability by default, that generally wins over flexibility. So the idea is to make the resource easier to deploy with WAF compliance than without.

However in this case I think we can achieve both. The implemented logic you mentioned should also work with the default value set to `true`, and that would be used by default unless the sku is set to free. What about adding more context in the enablePurgeProtection @description? Something like: `defaults to true unless 'sku' is set to 'Free', since purge protection is not available in Free tier.`

Reply:
Unresolved this, since purge protection enablement in test files is currently blocking the validation pipeline. My suggestion is still the one of the first comment of this thread:

- Disable the feature in all tests and add a comment next to the assignment mentioning that is not WAF compliant but only an exception for deployment validation
- Add an exception for this rule in the psrule setting file, as done for keyvault for the same reason https://github.com/Azure/bicep-registry-modules/blob/main/avm/utilities/pipelines/staticValidation/psrule/ps-rule.yaml#L78